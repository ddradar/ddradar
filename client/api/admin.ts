import type { NuxtHTTPInstance } from '@nuxt/http'

import type { SongInfo } from '~/api/song'
import type { NotificationSchema } from '~/core/db/notification'

type NotificationRequest = Omit<
  NotificationSchema,
  'id' | 'sender' | 'timeStamp'
> & {
  id?: string
  timeStamp?: number
}

/**
 * Call "Post Notification" API. (Admin only)
 * @see https://github.com/ddradar/ddradar/tree/master/api/postNotification
 */
export function postNotification(
  $http: Pick<NuxtHTTPInstance, '$post'>,
  body: NotificationRequest
) {
  return $http.$post<NotificationRequest>('/api/v1/admin/notification', {
    ...body,
    sender: 'SYSTEM',
  })
}

/**
 * Call "Post Song Information" API. (Admin only)
 * @see https://github.com/ddradar/ddradar/tree/master/api/postSongInfo
 */
export function postSongInfo(
  $http: Pick<NuxtHTTPInstance, '$post'>,
  body: SongInfo
) {
  return $http.$post<SongInfo>('/api/v1/admin/songs', body)
}
