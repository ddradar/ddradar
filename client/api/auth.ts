import type { Api } from '@ddradar/core'
import type { NuxtHTTPInstance } from '@nuxt/http'

type AuthResult = {
  clientPrincipal: Api.ClientPrincipal | null
}

/**
 * Call "Get Credentials" API.
 * @see https://docs.microsoft.com/azure/static-web-apps/user-information
 */
export async function getClientPrincipal(
  $http: Pick<NuxtHTTPInstance, '$get'>
) {
  const { clientPrincipal } = await $http.$get<AuthResult>('/.auth/me')
  return clientPrincipal
}
