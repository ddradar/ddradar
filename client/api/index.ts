export const apiPrefix = '/api/v1'
