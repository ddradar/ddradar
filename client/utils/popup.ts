import type { BuefyNamespace } from 'buefy/types'

const option = { position: 'is-top', hasIcon: true } as const

export function success(
  $buefy: Pick<BuefyNamespace, 'notification'>,
  message: string
) {
  $buefy.notification.open({ message, type: 'is-success', ...option })
}

export function warning(
  $buefy: Pick<BuefyNamespace, 'notification'>,
  message: string
) {
  $buefy.notification.open({ message, type: 'is-warning', ...option })
}

export function danger(
  $buefy: Pick<BuefyNamespace, 'notification'>,
  message: string
) {
  $buefy.notification.open({ message, type: 'is-danger', ...option })
}
