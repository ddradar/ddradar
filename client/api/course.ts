import type { Api } from '@ddradar/core'
import type { NuxtHTTPInstance } from '@nuxt/http'

import { apiPrefix } from '~/api'

export function getCourseType(type: number) {
  return type === 1 ? 'NONSTOP' : type === 2 ? '段位認定' : ''
}

/**
 * Call "Get Course List" API.
 * @see https://github.com/ddradar/ddradar/tree/master/api/courses/
 */
export function getCourseList(
  $http: Pick<NuxtHTTPInstance, '$get'>,
  series?: 16 | 17,
  type?: 1 | 2
) {
  const searchParams = new URLSearchParams()
  if (series) searchParams.append('series', `${series}`)
  if (type) searchParams.append('type', `${type}`)
  return $http.$get<Api.CourseListData[]>(`${apiPrefix}/courses`, {
    searchParams,
  })
}

/**
 * Call "Get Course Information" API.
 * @see https://github.com/ddradar/ddradar/tree/master/api/courses__id/
 */
export function getCourseInfo(
  $http: Pick<NuxtHTTPInstance, '$get'>,
  id: string
) {
  return $http.$get<Api.CourseInfo>(`${apiPrefix}/courses/${id}`)
}
