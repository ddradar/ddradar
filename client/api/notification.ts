import type { Api } from '@ddradar/core'
import type { NuxtHTTPInstance } from '@nuxt/http'

import { apiPrefix } from '~/api'

/**
 * Call "Get Notification List" API.
 * @see https://github.com/ddradar/ddradar/tree/master/api/notification/
 */
export function getNotificationList(
  $http: Pick<NuxtHTTPInstance, '$get'>,
  topOnly?: boolean
) {
  const query = topOnly ? '?scope=top' : ''
  return $http.$get<Api.NotificationListData[]>(
    `${apiPrefix}/notification${query}`
  )
}

/**
 * Call "Get Notification Info" API.
 * @see https://github.com/ddradar/ddradar/tree/master/api/notification__id/
 */
export function getNotificationInfo(
  $http: Pick<NuxtHTTPInstance, '$get'>,
  id: string
) {
  return $http.$get<Api.NotificationInfo>(`${apiPrefix}/notification/${id}`)
}

/**
 * Call "Post Notification" API. (Admin only)
 * @see https://github.com/ddradar/ddradar/tree/master/api/notification--post/
 */
export function postNotification(
  $http: Pick<NuxtHTTPInstance, '$post'>,
  body: Omit<Api.NotificationBody, 'sender'>
) {
  return $http.$post<Api.NotificationBody>(`${apiPrefix}/notification`, {
    ...body,
    sender: 'SYSTEM',
  })
}
