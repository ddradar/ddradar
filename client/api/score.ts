import type { Api } from '@ddradar/core'
import type { NuxtHTTPInstance } from '@nuxt/http'

import { apiPrefix } from '~/api'

/**
 * Call "Get Chart Score" API.
 * @see https://github.com/ddradar/ddradar/tree/master/api/scores__id__style__difficulty--get/
 */
export function getChartScore(
  $http: Pick<NuxtHTTPInstance, '$get'>,
  songId: string,
  playStyle: 1 | 2,
  difficulty: 0 | 1 | 2 | 3 | 4,
  scope?: 'private' | 'medium' | 'full'
) {
  const query = scope ? `?scope=${scope}` : ''
  return $http.$get<Api.ScoreInfo[]>(
    `${apiPrefix}/scores/${songId}/${playStyle}/${difficulty}${query}`
  )
}

/**
 * Call "Post Chart Score" API.
 * @see https://github.com/ddradar/ddradar/tree/master/api/scores__id__style__difficulty--post/
 */
export function postChartScore(
  $http: Pick<NuxtHTTPInstance, '$post'>,
  songId: string,
  playStyle: 1 | 2,
  difficulty: 0 | 1 | 2 | 3 | 4,
  score: Api.ScoreBody
) {
  return $http.$post(
    `${apiPrefix}/scores/${songId}/${playStyle}/${difficulty}`,
    score
  )
}

/**
 * Call "Delete Chart Score" API.
 * @see https://github.com/ddradar/ddradar/tree/master/api/scores__id__style__difficulty--delete/
 */
export function deleteChartScore(
  $http: Pick<NuxtHTTPInstance, 'delete'>,
  songId: string,
  playStyle: 1 | 2,
  difficulty: 0 | 1 | 2 | 3 | 4
) {
  return $http.delete(`/api/v1/scores/${songId}/${playStyle}/${difficulty}`)
}

/**
 * Call "Post Song Scores" API.
 * @see https://github.com/ddradar/ddradar/tree/master/api/scores__id--post/
 */
export function postSongScores(
  $http: Pick<NuxtHTTPInstance, '$post'>,
  songId: string,
  scores: Api.ScoreListBody[]
) {
  return $http.$post<Api.ScoreInfo[]>(`${apiPrefix}/scores/${songId}`, scores)
}
