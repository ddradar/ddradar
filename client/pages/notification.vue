<template>
  <section class="section">
    <h1 class="title">{{ $t('pageTitle') }}</h1>
    <section>
      <b-table
        :data="messages"
        paginated
        per-page="20"
        striped
        :loading="$fetchState.pending"
        detailed
        detail-key="id"
        show-detail-icon
      >
        <b-table-column v-slot="props" field="icon">
          <b-button :type="props.row.type" :icon-left="props.row.icon" />
        </b-table-column>
        <b-table-column v-slot="props" field="title" :label="$t('title')">
          {{ props.row.title }}
        </b-table-column>
        <b-table-column v-slot="props" field="timeStamp" :label="$t('date')">
          {{ props.row.date }}
        </b-table-column>
        <b-table-column
          v-slot="props"
          :visible="$accessor.isAdmin"
          label="Edit"
        >
          <nuxt-link :to="`/admin/notification/${props.row.id}`">
            <b-icon icon="pencil-box-outline" />
          </nuxt-link>
        </b-table-column>

        <template #detail="props">
          <!-- eslint-disable-next-line vue/no-v-html -->
          <div class="content" v-html="props.row.body" />
        </template>

        <template #empty>
          <section v-if="$fetchState.pending" class="section">
            <b-skeleton animated />
            <b-skeleton animated />
            <b-skeleton animated />
          </section>
          <section v-else class="section">
            <div class="content has-text-grey has-text-centered">
              <p>{{ $t('noData') }}</p>
            </div>
          </section>
        </template>
      </b-table>
    </section>
  </section>
</template>

<i18n>
{
  "ja": {
    "pageTitle": "お知らせ一覧",
    "title": "タイトル",
    "date": "投稿日時",
    "noData": "データがありません"
  },
  "en": {
    "pageTitle": "All Notification List",
    "title": "Title",
    "date": "Date",
    "noData": "No data"
  }
}
</i18n>

<script lang="ts">
import type { Api } from '@ddradar/core'
import marked from 'marked'
import { Component, Vue } from 'nuxt-property-decorator'
import type { MetaInfo } from 'vue-meta'

import { getNotificationList } from '~/api/notification'
import { unixTimeToString } from '~/utils/date'
import * as popup from '~/utils/popup'

type NotificationDetail = Omit<Api.NotificationListData, 'timeStamp'> & {
  date: string
}

@Component({ fetchOnServer: false })
export default class UserListPage extends Vue {
  messages: NotificationDetail[] = []

  /* istanbul ignore next */
  head(): MetaInfo {
    return {
      title: this.$t('pageTitle') as string,
    }
  }

  /** Call "Get Notification API" once. */
  async fetch() {
    try {
      const messages = await getNotificationList(this.$http)
      this.messages = messages.map(d => ({
        id: d.id,
        type: d.type,
        icon: d.icon,
        title: d.title,
        body: marked(d.body),
        date: unixTimeToString(d.timeStamp),
      }))
    } catch (error) {
      popup.danger(this.$buefy, error.message ?? error)
    }
  }
}
</script>
