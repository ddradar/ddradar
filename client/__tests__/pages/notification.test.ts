import { mount, RouterLinkStub, shallowMount, Wrapper } from '@vue/test-utils'

import { notificationList } from '~/__tests__/data'
import { createI18n, createVue } from '~/__tests__/util'
import { getNotificationList } from '~/api/notification'
import Page from '~/pages/notification.vue'
import * as popup from '~/utils/popup'

jest.mock('~/api/notification')
jest.mock('~/utils/popup')

const localVue = createVue()

describe('/pages/notification.vue', () => {
  const messages = notificationList.map(d => ({
    ...d,
    date: '2020/8/13 0:00:00',
  }))
  const $accessor = { isAdmin: false }
  const $fetchState = { pending: false }

  describe.each(['en', 'ja'])('{ locale: "%s" } snapshot test', locale => {
    const i18n = createI18n(locale)
    const stubs = { NuxtLink: RouterLinkStub }
    test('renders loading skeleton', async () => {
      // Arrange
      const mocks = { $fetchState: { pending: true }, $accessor }
      const data = () => ({ messages: [] })

      // Act
      const wrapper = mount(Page, { localVue, mocks, stubs, data, i18n })
      await wrapper.vm.$nextTick()

      // Assert
      expect(wrapper).toMatchSnapshot()
    })
    test('renders correctly', async () => {
      // Arrange
      const mocks = { $fetchState, $accessor }
      const data = () => ({ messages })

      // Act
      const wrapper = mount(Page, { localVue, mocks, stubs, data, i18n })
      await wrapper.vm.$nextTick()

      // Assert
      expect(wrapper).toMatchSnapshot()
    })
    test('renders empty', async () => {
      // Arrange
      const mocks = { $fetchState, $accessor }
      const data = () => ({ messages: [] })

      // Act
      const wrapper = mount(Page, { localVue, mocks, stubs, data, i18n })
      await wrapper.vm.$nextTick()

      // Assert
      expect(wrapper).toMatchSnapshot()
    })
    test('renders Edit column if admin', async () => {
      // Arrange
      const mocks = { $fetchState, $accessor: { isAdmin: true } }
      const data = () => ({ messages })

      // Act
      const wrapper = mount(Page, { localVue, mocks, stubs, data, i18n })
      await wrapper.vm.$nextTick()

      // Assert
      expect(wrapper).toMatchSnapshot()
    })
  })

  // Lifecycle
  describe('fetch()', () => {
    let wrapper: Wrapper<Page>
    const i18n = createI18n()
    const mocks = { $accessor, $buefy: {}, $fetchState, $http: {} }
    const data = () => ({ messages: [] })
    beforeEach(() => {
      jest.mocked(getNotificationList).mockClear()
      jest.mocked(popup.danger).mockClear()
      wrapper = shallowMount(Page, { localVue, mocks, data, i18n })
    })

    test('calls getNotificationList($http)', async () => {
      // Arrange
      jest.mocked(getNotificationList).mockResolvedValue(notificationList)

      // Act
      // @ts-ignore
      await wrapper.vm.$options.fetch?.call(wrapper.vm)

      // Assert
      expect(jest.mocked(getNotificationList)).toBeCalledTimes(1)
      expect(jest.mocked(getNotificationList)).toBeCalledWith(mocks.$http)
      expect(jest.mocked(popup.danger)).not.toBeCalled()
      expect(wrapper.vm.$data.messages).toHaveLength(2)
    })
    test('calls popup.danger() if getNotificationList($http) throws error', async () => {
      // Arrange
      const errorMessage = 'invalid'
      jest.mocked(getNotificationList).mockRejectedValue(errorMessage)

      // Act
      // @ts-ignore
      await wrapper.vm.$options.fetch?.call(wrapper.vm)

      // Assert
      expect(jest.mocked(getNotificationList)).toBeCalledTimes(1)
      expect(jest.mocked(getNotificationList)).toBeCalledWith(mocks.$http)
      expect(jest.mocked(popup.danger)).toBeCalledTimes(1)
      expect(jest.mocked(popup.danger)).toBeCalledWith(
        mocks.$buefy,
        errorMessage
      )
      expect(wrapper.vm.$data.messages).toHaveLength(0)
    })
  })
})
