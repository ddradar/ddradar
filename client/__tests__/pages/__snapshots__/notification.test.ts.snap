// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/pages/notification.vue snapshot test renders Edit column if admin 1`] = `
<section class="section">
  <h1 class="title">お知らせ一覧</h1>
  <section>
    <div class="b-table">
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <div class="table-wrapper has-mobile-cards">
        <table class="table is-striped">
          <thead>
            <tr>
              <th width="40px"></th>
              <!---->
              <th draggable="false" class="">
                <div class="th-wrap"><span class="is-relative">  <span class="icon sort-icon is-small is-invisible"><i class="mdi mdi-arrow-up"></i></span></span></div>
              </th>
              <th draggable="false" class="">
                <div class="th-wrap"><span class="is-relative"> タイトル <span class="icon sort-icon is-small is-invisible"><i class="mdi mdi-arrow-up"></i></span></span></div>
              </th>
              <th draggable="false" class="">
                <div class="th-wrap"><span class="is-relative"> 投稿日時 <span class="icon sort-icon is-small is-invisible"><i class="mdi mdi-arrow-up"></i></span></span></div>
              </th>
              <th draggable="false" class="">
                <div class="th-wrap"><span class="is-relative"> Edit <span class="icon sort-icon is-small is-invisible"><i class="mdi mdi-arrow-up"></i></span></span></div>
              </th>
              <!---->
            </tr>
            <!---->
            <!---->
          </thead>
          <tbody>
            <tr draggable="false" class="">
              <td class="chevron-cell"><a role="button"><span class="icon"><i class="mdi mdi-chevron-right mdi-24px"></i></span></a></td>
              <!---->
              <td class=""><button type="button" class="button is-info"><span class="icon is-small"><i class="mdi mdi-account"></i></span>
                  <!---->
                  <!---->
                </button></td>
              <td data-label="タイトル" class="">
                Title 1
              </td>
              <td data-label="投稿日時" class="">
                2020/8/13 0:00:00
              </td>
              <td data-label="Edit" class=""><a><span class="icon"><i class="mdi mdi-pencil-box-outline mdi-24px"></i></span></a></td>
              <!---->
            </tr>
            <transition-stub name="">
              <!---->
            </transition-stub>
            <!---->
            <tr draggable="false" class="">
              <td class="chevron-cell"><a role="button"><span class="icon"><i class="mdi mdi-chevron-right mdi-24px"></i></span></a></td>
              <!---->
              <td class=""><button type="button" class="button is-info">
                  <!---->
                  <!---->
                  <!---->
                </button></td>
              <td data-label="タイトル" class="">
                Title 2
              </td>
              <td data-label="投稿日時" class="">
                2020/8/13 0:00:00
              </td>
              <td data-label="Edit" class=""><a><span class="icon"><i class="mdi mdi-pencil-box-outline mdi-24px"></i></span></a></td>
              <!---->
            </tr>
            <transition-stub name="">
              <!---->
            </transition-stub>
            <!---->
            <!---->
          </tbody>
          <!---->
        </table>
        <!---->
      </div>
      <div class="top level">
        <div class="level-left"></div>
        <div class="level-right">
          <div class="level-item">
            <nav class="pagination"><a role="button" href="#" disabled="disabled" class="pagination-link pagination-previous"><span class="icon" aria-hidden="true"><i class="mdi mdi-chevron-left mdi-24px"></i></span></a><a role="button" href="#" disabled="disabled" class="pagination-link pagination-next"><span class="icon" aria-hidden="true"><i class="mdi mdi-chevron-right mdi-24px"></i></span></a>
              <ul class="pagination-list">
                <!---->
                <!---->
                <li><a role="button" href="#" aria-current="true" class="pagination-link is-current ">1</a></li>
                <!---->
                <!---->
              </ul>
            </nav>
          </div>
        </div>
      </div>
    </div>
  </section>
</section>
`;

exports[`/pages/notification.vue snapshot test renders correctly if { locale: en } 1`] = `
<section class="section">
  <h1 class="title">All Notification List</h1>
  <section>
    <div class="b-table">
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <div class="table-wrapper has-mobile-cards">
        <table class="table is-striped">
          <thead>
            <tr>
              <th width="40px"></th>
              <!---->
              <th draggable="false" class="">
                <div class="th-wrap"><span class="is-relative">  <span class="icon sort-icon is-small is-invisible"><i class="mdi mdi-arrow-up"></i></span></span></div>
              </th>
              <th draggable="false" class="">
                <div class="th-wrap"><span class="is-relative"> Title <span class="icon sort-icon is-small is-invisible"><i class="mdi mdi-arrow-up"></i></span></span></div>
              </th>
              <th draggable="false" class="">
                <div class="th-wrap"><span class="is-relative"> Date <span class="icon sort-icon is-small is-invisible"><i class="mdi mdi-arrow-up"></i></span></span></div>
              </th>
              <!---->
            </tr>
            <!---->
            <!---->
          </thead>
          <tbody>
            <tr draggable="false" class="">
              <td class="chevron-cell"><a role="button"><span class="icon"><i class="mdi mdi-chevron-right mdi-24px"></i></span></a></td>
              <!---->
              <td class=""><button type="button" class="button is-info"><span class="icon is-small"><i class="mdi mdi-account"></i></span>
                  <!---->
                  <!---->
                </button></td>
              <td data-label="Title" class="">
                Title 1
              </td>
              <td data-label="Date" class="">
                2020/8/13 0:00:00
              </td>
              <!---->
            </tr>
            <transition-stub name="">
              <!---->
            </transition-stub>
            <!---->
            <tr draggable="false" class="">
              <td class="chevron-cell"><a role="button"><span class="icon"><i class="mdi mdi-chevron-right mdi-24px"></i></span></a></td>
              <!---->
              <td class=""><button type="button" class="button is-info">
                  <!---->
                  <!---->
                  <!---->
                </button></td>
              <td data-label="Title" class="">
                Title 2
              </td>
              <td data-label="Date" class="">
                2020/8/13 0:00:00
              </td>
              <!---->
            </tr>
            <transition-stub name="">
              <!---->
            </transition-stub>
            <!---->
            <!---->
          </tbody>
          <!---->
        </table>
        <!---->
      </div>
      <div class="top level">
        <div class="level-left"></div>
        <div class="level-right">
          <div class="level-item">
            <nav class="pagination"><a role="button" href="#" disabled="disabled" class="pagination-link pagination-previous"><span class="icon" aria-hidden="true"><i class="mdi mdi-chevron-left mdi-24px"></i></span></a><a role="button" href="#" disabled="disabled" class="pagination-link pagination-next"><span class="icon" aria-hidden="true"><i class="mdi mdi-chevron-right mdi-24px"></i></span></a>
              <ul class="pagination-list">
                <!---->
                <!---->
                <li><a role="button" href="#" aria-current="true" class="pagination-link is-current ">1</a></li>
                <!---->
                <!---->
              </ul>
            </nav>
          </div>
        </div>
      </div>
    </div>
  </section>
</section>
`;

exports[`/pages/notification.vue snapshot test renders correctly if { locale: ja } 1`] = `
<section class="section">
  <h1 class="title">お知らせ一覧</h1>
  <section>
    <div class="b-table">
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <div class="table-wrapper has-mobile-cards">
        <table class="table is-striped">
          <thead>
            <tr>
              <th width="40px"></th>
              <!---->
              <th draggable="false" class="">
                <div class="th-wrap"><span class="is-relative">  <span class="icon sort-icon is-small is-invisible"><i class="mdi mdi-arrow-up"></i></span></span></div>
              </th>
              <th draggable="false" class="">
                <div class="th-wrap"><span class="is-relative"> タイトル <span class="icon sort-icon is-small is-invisible"><i class="mdi mdi-arrow-up"></i></span></span></div>
              </th>
              <th draggable="false" class="">
                <div class="th-wrap"><span class="is-relative"> 投稿日時 <span class="icon sort-icon is-small is-invisible"><i class="mdi mdi-arrow-up"></i></span></span></div>
              </th>
              <!---->
            </tr>
            <!---->
            <!---->
          </thead>
          <tbody>
            <tr draggable="false" class="">
              <td class="chevron-cell"><a role="button"><span class="icon"><i class="mdi mdi-chevron-right mdi-24px"></i></span></a></td>
              <!---->
              <td class=""><button type="button" class="button is-info"><span class="icon is-small"><i class="mdi mdi-account"></i></span>
                  <!---->
                  <!---->
                </button></td>
              <td data-label="タイトル" class="">
                Title 1
              </td>
              <td data-label="投稿日時" class="">
                2020/8/13 0:00:00
              </td>
              <!---->
            </tr>
            <transition-stub name="">
              <!---->
            </transition-stub>
            <!---->
            <tr draggable="false" class="">
              <td class="chevron-cell"><a role="button"><span class="icon"><i class="mdi mdi-chevron-right mdi-24px"></i></span></a></td>
              <!---->
              <td class=""><button type="button" class="button is-info">
                  <!---->
                  <!---->
                  <!---->
                </button></td>
              <td data-label="タイトル" class="">
                Title 2
              </td>
              <td data-label="投稿日時" class="">
                2020/8/13 0:00:00
              </td>
              <!---->
            </tr>
            <transition-stub name="">
              <!---->
            </transition-stub>
            <!---->
            <!---->
          </tbody>
          <!---->
        </table>
        <!---->
      </div>
      <div class="top level">
        <div class="level-left"></div>
        <div class="level-right">
          <div class="level-item">
            <nav class="pagination"><a role="button" href="#" disabled="disabled" class="pagination-link pagination-previous"><span class="icon" aria-hidden="true"><i class="mdi mdi-chevron-left mdi-24px"></i></span></a><a role="button" href="#" disabled="disabled" class="pagination-link pagination-next"><span class="icon" aria-hidden="true"><i class="mdi mdi-chevron-right mdi-24px"></i></span></a>
              <ul class="pagination-list">
                <!---->
                <!---->
                <li><a role="button" href="#" aria-current="true" class="pagination-link is-current ">1</a></li>
                <!---->
                <!---->
              </ul>
            </nav>
          </div>
        </div>
      </div>
    </div>
  </section>
</section>
`;

exports[`/pages/notification.vue snapshot test renders empty if { locale: en } 1`] = `
<section class="section">
  <h1 class="title">All Notification List</h1>
  <section>
    <div class="b-table">
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <div class="table-wrapper has-mobile-cards">
        <table class="table is-striped">
          <thead>
            <tr>
              <th width="40px"></th>
              <!---->
              <th draggable="false" class="">
                <div class="th-wrap"><span class="is-relative">  <span class="icon sort-icon is-small is-invisible"><i class="mdi mdi-arrow-up"></i></span></span></div>
              </th>
              <th draggable="false" class="">
                <div class="th-wrap"><span class="is-relative"> Title <span class="icon sort-icon is-small is-invisible"><i class="mdi mdi-arrow-up"></i></span></span></div>
              </th>
              <th draggable="false" class="">
                <div class="th-wrap"><span class="is-relative"> Date <span class="icon sort-icon is-small is-invisible"><i class="mdi mdi-arrow-up"></i></span></span></div>
              </th>
              <!---->
            </tr>
            <!---->
            <!---->
          </thead>
          <tbody>
            <tr class="is-empty">
              <td colspan="4">
                <section class="section">
                  <div class="content has-text-grey has-text-centered">
                    <p>No data</p>
                  </div>
                </section>
              </td>
            </tr>
          </tbody>
          <!---->
        </table>
        <!---->
      </div>
      <div class="top level">
        <div class="level-left"></div>
        <div class="level-right">
          <div class="level-item">
            <nav class="pagination"><a role="button" href="#" disabled="disabled" class="pagination-link pagination-previous"><span class="icon" aria-hidden="true"><i class="mdi mdi-chevron-left mdi-24px"></i></span></a><a role="button" href="#" disabled="disabled" class="pagination-link pagination-next"><span class="icon" aria-hidden="true"><i class="mdi mdi-chevron-right mdi-24px"></i></span></a>
              <ul class="pagination-list">
                <!---->
                <!---->
                <!---->
                <!---->
              </ul>
            </nav>
          </div>
        </div>
      </div>
    </div>
  </section>
</section>
`;

exports[`/pages/notification.vue snapshot test renders empty if { locale: ja } 1`] = `
<section class="section">
  <h1 class="title">お知らせ一覧</h1>
  <section>
    <div class="b-table">
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <div class="table-wrapper has-mobile-cards">
        <table class="table is-striped">
          <thead>
            <tr>
              <th width="40px"></th>
              <!---->
              <th draggable="false" class="">
                <div class="th-wrap"><span class="is-relative">  <span class="icon sort-icon is-small is-invisible"><i class="mdi mdi-arrow-up"></i></span></span></div>
              </th>
              <th draggable="false" class="">
                <div class="th-wrap"><span class="is-relative"> タイトル <span class="icon sort-icon is-small is-invisible"><i class="mdi mdi-arrow-up"></i></span></span></div>
              </th>
              <th draggable="false" class="">
                <div class="th-wrap"><span class="is-relative"> 投稿日時 <span class="icon sort-icon is-small is-invisible"><i class="mdi mdi-arrow-up"></i></span></span></div>
              </th>
              <!---->
            </tr>
            <!---->
            <!---->
          </thead>
          <tbody>
            <tr class="is-empty">
              <td colspan="4">
                <section class="section">
                  <div class="content has-text-grey has-text-centered">
                    <p>データがありません</p>
                  </div>
                </section>
              </td>
            </tr>
          </tbody>
          <!---->
        </table>
        <!---->
      </div>
      <div class="top level">
        <div class="level-left"></div>
        <div class="level-right">
          <div class="level-item">
            <nav class="pagination"><a role="button" href="#" disabled="disabled" class="pagination-link pagination-previous"><span class="icon" aria-hidden="true"><i class="mdi mdi-chevron-left mdi-24px"></i></span></a><a role="button" href="#" disabled="disabled" class="pagination-link pagination-next"><span class="icon" aria-hidden="true"><i class="mdi mdi-chevron-right mdi-24px"></i></span></a>
              <ul class="pagination-list">
                <!---->
                <!---->
                <!---->
                <!---->
              </ul>
            </nav>
          </div>
        </div>
      </div>
    </div>
  </section>
</section>
`;

exports[`/pages/notification.vue snapshot test renders loading skeleton 1`] = `
<section class="section">
  <h1 class="title">お知らせ一覧</h1>
  <section>
    <div class="b-table">
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <div class="table-wrapper has-mobile-cards">
        <table class="table is-striped">
          <thead>
            <tr>
              <th width="40px"></th>
              <!---->
              <th draggable="false" class="">
                <div class="th-wrap"><span class="is-relative">  <span class="icon sort-icon is-small is-invisible"><i class="mdi mdi-arrow-up"></i></span></span></div>
              </th>
              <th draggable="false" class="">
                <div class="th-wrap"><span class="is-relative"> タイトル <span class="icon sort-icon is-small is-invisible"><i class="mdi mdi-arrow-up"></i></span></span></div>
              </th>
              <th draggable="false" class="">
                <div class="th-wrap"><span class="is-relative"> 投稿日時 <span class="icon sort-icon is-small is-invisible"><i class="mdi mdi-arrow-up"></i></span></span></div>
              </th>
              <!---->
            </tr>
            <!---->
            <!---->
          </thead>
          <tbody>
            <tr class="is-empty">
              <td colspan="4">
                <section class="section">
                  <div class="b-skeleton is-animated">
                    <div class="b-skeleton-item is-rounded"></div>
                  </div>
                  <div class="b-skeleton is-animated">
                    <div class="b-skeleton-item is-rounded"></div>
                  </div>
                  <div class="b-skeleton is-animated">
                    <div class="b-skeleton-item is-rounded"></div>
                  </div>
                </section>
              </td>
            </tr>
          </tbody>
          <!---->
        </table>
        <transition-stub name="fade">
          <div class="loading-overlay is-active">
            <div class="loading-background"></div>
            <div class="loading-icon"></div>
          </div>
        </transition-stub>
      </div>
      <div class="top level">
        <div class="level-left"></div>
        <div class="level-right">
          <div class="level-item">
            <nav class="pagination"><a role="button" href="#" disabled="disabled" class="pagination-link pagination-previous"><span class="icon" aria-hidden="true"><i class="mdi mdi-chevron-left mdi-24px"></i></span></a><a role="button" href="#" disabled="disabled" class="pagination-link pagination-next"><span class="icon" aria-hidden="true"><i class="mdi mdi-chevron-right mdi-24px"></i></span></a>
              <ul class="pagination-list">
                <!---->
                <!---->
                <!---->
                <!---->
              </ul>
            </nav>
          </div>
        </div>
      </div>
    </div>
  </section>
</section>
`;
